src/components/SearchBar.js

```diff
@@ -1,15 +1,46 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { searchPhoto, onType } from '../actions';
 
-const SearchBar = () => {
-    return (
-        <div className="ui loading search">
-            <div className="ui icon input">
-                <input className="prompt" type="text" placeholder="Search..." />
-                <i className="search icon"></i>
-            </div>
-            <div className="results"></div>
-        </div>
-    );
+class SearchBar extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    onSearchChange = (event) => {
+        event.preventDefault();
+
+        this.props.onType(event.target.value);
+    }
+
+    onSearchSubmit = (event) => {
+        event.preventDefault();
+
+        this.props.searchPhoto(this.props.searchTerm);
+    }
+
+    render() {
+        return (
+            <form className="ui medium">
+                <div className="ui fluid search">
+                    <div className="ui icon input">
+                        <input 
+                            className="prompt" 
+                            type="text" 
+                            placeholder="Search..." 
+                            onChange={this.onSearchChange}
+                            value={this.props.searchTerm}/>
+                        <i className="search icon"></i>
+                    </div>
+                    <div className="results"></div>
+                </div>
+            </form>
+        );
+    }
+}
+
+const mapStateToProps = (state) => {
+    return { photos: state.photos, searchTerm: state.searchTerm };
 }
 
-export default SearchBar;
\ No newline at end of file
+export default connect(mapStateToProps, { searchPhoto, onType })(SearchBar);
```

Comment:
Please define a mapDistpatch function rather than just inline definition. This looks fine for now. But if we think about this is in the long term, imagine having 20 dispatch functions, this can get to be a bit much. It also helps with readability.

Reply:
https://github.com/Shenanigans510/yeet/pull/2/commits/4436147150102641f09f1e990150b1db11f39dac